src/extension.ts

```diff
@@ -15,66 +16,112 @@ export function activate(ctx: vscode.ExtensionContext) {
+      endpoints.showPreviousAnswer(snippetsStorage)
+    )
   );
-  vscode.commands.registerCommand("snippet.showNextAnswer", () =>
-    endpoints.showNextAnswer(snippetsStorage)
+  ctx.subscriptions.push(
+    vscode.commands.registerCommand("snippet.showNextAnswer", () =>
+      endpoints.showNextAnswer(snippetsStorage)
+    )
   );
-  vscode.commands.registerCommand(
-    "snippet.toggleComments",
-    endpoints.toggleComments
+  ctx.subscriptions.push(
+    vscode.commands.registerCommand(
+      "snippet.toggleComments",
+      endpoints.toggleComments
+    )
   );
 
-  vscode.commands.registerCommand(
-    "snippet.saveSnippet",
-    endpoints.saveSnippet(snippetsTreeProvider)
+  ctx.subscriptions.push(
+    vscode.commands.registerCommand(
+      "snippet.saveSnippet",
+      endpoints.saveSnippet(snippetsTreeProvider)
+    )
   );
-  vscode.commands.registerCommand(
-    "snippet.insertSnippet",
-    endpoints.insertSnippet(snippetsTreeProvider)
+  ctx.subscriptions.push(
+    vscode.commands.registerCommand(
+      "snippet.insertSnippet",
+      endpoints.insertSnippet(snippetsTreeProvider)
+    )
   );
-  vscode.commands.registerCommand(
-    "snippet.deleteSnippet",
-    endpoints.deleteSnippet(snippetsTreeProvider)
+  ctx.subscriptions.push(
+    vscode.commands.registerCommand(
+      "snippet.deleteSnippet",
+      endpoints.deleteSnippet(snippetsTreeProvider)
+    )
   );
-  vscode.commands.registerCommand(
-    "snippet.renameSnippet",
-    endpoints.renameSnippet(snippetsTreeProvider)
+  ctx.subscriptions.push(
+    vscode.commands.registerCommand(
+      "snippet.renameSnippet",
+      endpoints.renameSnippet(snippetsTreeProvider)
+    )
   );
-  vscode.commands.registerCommand(
-    "snippet.copySnippet",
-    endpoints.copySnippet(snippetsTreeProvider)
+  ctx.subscriptions.push(
+    vscode.commands.registerCommand(
+      "snippet.copySnippet",
+      endpoints.copySnippet(snippetsTreeProvider)
+    )
   );
-  vscode.commands.registerCommand(
-    "snippet.findAndCopy",
-    endpoints.findAndCopy(snippetsStorage)
+  ctx.subscriptions.push(
+    vscode.commands.registerCommand(
+      "snippet.findAndCopy",
+      endpoints.findAndCopy(snippetsStorage)
+    )
   );
-  vscode.commands.registerCommand(
-    "snippet.createFolder",
-    endpoints.createFolder(snippetsTreeProvider)
+  ctx.subscriptions.push(
+    vscode.commands.registerCommand(
+      "snippet.createFolder",
+      endpoints.createFolder(snippetsTreeProvider)
+    )
   );
+  ctx.subscriptions.push(
+    vscode.commands.registerCommand(
+      "snippet.restoreBackups",
+      endpoints.showBackups(backupManager)
+    )
+  );
+  if (process.env.NODE_ENV === "test") {
+    ctx.subscriptions.push(
+      vscode.commands.registerCommand(
+        "snippet.test_moveElement",
+        async (sourceId, targetId) =>
+          await snippetsStorage.moveElement(sourceId, targetId)
+      )
+    );
+  }
```

Comment:
Were you planning to keep this?

Reply:
Yes, so the tests for the backup system check that when we `rename`, `save`, `delete` or `move` snippets.
`rename`, `save` and `delete` are exposed as commands, so we can call them in tests (we don't directly instantiate different classes such as `BackupManager` and don't call their API in tests, but instead interact with the extension via commands, like the real user will do). 
But the `move` is executed only via drag-and-drop (in the `SnippetsTreeProvider` class, the `handleDrop` method calls `storage.moveElement`). So, we need to call `storage.moveElement` in tests but it is not exposed outside, it can only be called via `handleDrop`, and I haven't found how to call it in tests. 

So I decided to expose `storage.moveElement` as a command (but only in test environment: `process.env.NODE_ENV === "test"`). This looks like a hack but I haven't found a better solution